src/cromshell/cost/command.py

```diff
@@ -0,0 +1,302 @@
+    ):
+        LOGGER.error("Could not find workflow id %s in submission file.", resolved_workflow_id)
+        raise ValueError(
+            f"Could not find workflow id {resolved_workflow_id} in submission file."
+        )
+
+    # check that bq setting in json config
+    check_cost_table_is_configured(config_options=config.cromshell_config_options)
+
+    # Get time workflow finished using metadata command (error if not finished)
+    LOGGER.info("Retrieving workflow metadata")
+    workflow_metadata = get_metadata(config=config)
+    start_time, end_time = get_submission_start_end_time(workflow_metadata=workflow_metadata)
+
+    LOGGER.info("Checking workflow completed and finished past 24hrs")
+    checks_before_query(end_time=end_time, workflow_id=resolved_workflow_id)
+
+    LOGGER.info("Querying BQ")
+    # Create start and end time for query, plus/minus a day from start and finish
+    query_start_time = str(datetime.strptime(start_time, "%Y-%m-%dT%H:%M:%S.%f%z") - timedelta(days=1))
+    query_end_time = str(datetime.strptime(end_time, "%Y-%m-%dT%H:%M:%S.%f%z") + timedelta(days=1))
+
+    query_results = query_bigquery(
+        workflow_id=resolved_workflow_id,
+        bq_database=config.cromshell_config_options["bq_cost_table"],
+        start_date=query_start_time,
+        end_date=query_end_time,
+        detailed=detailed
+    )
+
+    temp_query_result_csv = NamedTemporaryFile()
+    write_bq_results_to_temp_csv(
+        tempfile=temp_query_result_csv, results=query_results, detailed=detailed
+    )
+
+    temp_query_result_csv_rounded = NamedTemporaryFile()
+    round_cost_values(
+        query_result_csv=temp_query_result_csv,
+        query_result_csv_rounded=temp_query_result_csv_rounded,
+        detailed=detailed
+    )
+
+    with open(temp_query_result_csv_rounded.name, 'r') as f:
+        print(f.read())
+        print("Costs rounded to nearest cent (approximately).")
+
+
+def print_query_results(query_results, detailed: bool) -> None:
+    # performing a step like getting the first row from the table would require to convert
+    # the table to datafram or CSV. Instead, this loop will get the first row and break.
+
+    if detailed:
+        for row in query_results:
+            print("{}\t{}\t{}\t{}".format(row.value, row.description, row.task_name,
+                                              row.cost))
+
+    else:
+        for row in query_results:
+            total_cost = round(float(row.cost), 2)
+            print(f"${total_cost}")
+            print("Costs rounded to nearest cent (approximately).")
+
+
+def query_bigquery(
+    workflow_id: str,
+    bq_database: str,
+    start_date: str,
+    end_date: str,
+    detailed: bool
+):
+    """
+        Query bigqueury for cost
+
+    :param workflow_id:
+    :param bq_database:
+    :param start_date:
+    :param end_date:
+    :return:
+    """
+    from google.cloud import bigquery
```

Comment:
Is it considered good practice to do imports as locally as possible?  It always seems weird to me because it means it's hard to tell what a file depends on and it won't fail for missing imports until later but maybe that's part of the python style?

Reply:
> import statements can be executed just about anywhere. It's often useful to place them inside functions to restrict their visibility and/or reduce initial startup time. Although Python's interpreter is optimized to not import the same module multiple times, repeatedly executing an import statement can seriously affect performance in some circumstances.

I was thinking local would be savvy because we could do some checks (has it been 24 hours, does workflow id exist) before importing the library. But I can move it back up to the top for visibility, I don't think its that big of an import.